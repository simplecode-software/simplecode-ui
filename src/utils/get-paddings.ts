import React from 'react';
import createCachedSelector from 're-reselect';

import { getCssSides } from './get-css-sides';


export const getPaddings = createCachedSelector<
    string | number | null,
    string | number | null,
    React.CSSProperties | null
>(
    (value) => value,
    (value) => getCssSides('padding', value),
)((value) => `getPaddings-${value}`);
