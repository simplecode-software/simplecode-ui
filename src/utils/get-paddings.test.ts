import { getPaddings } from './get-paddings';


describe('Get Paddings', () => {
    it('is a function', () => {
        expect(typeof getPaddings).toBe('function');
    });
    
    it('computes single paddings', () => {
        expect(getPaddings('10')).toMatchSnapshot();
    });
    
    it('computes double paddings', () => {
        expect(getPaddings('10 20')).toMatchSnapshot();
    });
    
    it('computes tripple paddings', () => {
        expect(getPaddings('10 20 30')).toMatchSnapshot();
    });
    
    it('computes full paddings', () => {
        expect(getPaddings('10 20 30 40')).toMatchSnapshot();
    });
    
    it('trims extra paddings', () => {
        expect(getPaddings('10 20 30 40 50')).toMatchSnapshot();
    });
    
    it('computes number padding', () => {
        expect(getPaddings(40)).toMatchSnapshot();
    });
    
    it('returns null if no padding', () => {
        expect(getPaddings(null)).toMatchSnapshot();
    });
    
    it('returns null if empty string', () => {
        expect(getPaddings('')).toMatchSnapshot();
    });
});
