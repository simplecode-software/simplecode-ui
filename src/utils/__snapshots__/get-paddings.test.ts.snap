// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Paddings computes double paddings 1`] = `
{
  "paddingBottom": 10,
  "paddingLeft": 20,
  "paddingRight": 20,
  "paddingTop": 10,
}
`;

exports[`Get Paddings computes full paddings 1`] = `
{
  "paddingBottom": 30,
  "paddingLeft": 40,
  "paddingRight": 20,
  "paddingTop": 10,
}
`;

exports[`Get Paddings computes number padding 1`] = `
{
  "padding": 40,
}
`;

exports[`Get Paddings computes single paddings 1`] = `
{
  "paddingBottom": 10,
  "paddingLeft": 10,
  "paddingRight": 10,
  "paddingTop": 10,
}
`;

exports[`Get Paddings computes tripple paddings 1`] = `
{
  "paddingBottom": 30,
  "paddingLeft": 20,
  "paddingRight": 20,
  "paddingTop": 10,
}
`;

exports[`Get Paddings returns null if empty string 1`] = `null`;

exports[`Get Paddings returns null if no padding 1`] = `null`;

exports[`Get Paddings trims extra paddings 1`] = `
{
  "paddingBottom": 30,
  "paddingLeft": 40,
  "paddingRight": 20,
  "paddingTop": 10,
}
`;
